datalad/metadata/parsers/fileinfo.py

```diff
@@ -0,0 +1,52 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Parser for generic file-based information (mime types, etc...)
+"""
+
+from datalad.metadata import _get_base_metadata_dict
+from datalad.support.annexrepo import AnnexRepo
+from datalad.metadata.parsers.base import BaseMetadataParser
+
+
+class MetadataParser(BaseMetadataParser):
+    _metadata_compliance = "http://docs.datalad.org/metadata.html#v0-1"
+    cfg_section = 'datalad.metadata.parser.fileinfo.report'
+
+    def has_metadata(self):
+        # could check if there is at least one annexed file, but hey...
+        repo = self.ds.repo
+        return repo and isinstance(repo, AnnexRepo)
+
+    def get_metadata(self, dsid=None, full=False):
+        meta = []
+        ds_meta = _get_base_metadata_dict(dsid)
+        parts = []
+        if not self.has_metadata():
+            return meta
+        # shortcuts
+        cfg = self.ds.config
+        repo = self.ds.repo
+        cfg_section = self.cfg_section
+        files = repo.get_annexed_files()
+        # TODO RF to do this with one annex call
```

Comment:
Note: For this TODO, an issue would be nice, telling that we need AnnexRepo.get_file_key() to work on a list and in "batch mode".